NSSServices/Controllers/VariablesController.cs

```diff
@@ -64,14 +69,136 @@ public override async Task<IActionResult> Get(int id)
             try
             {
                 if(id<0) return new BadRequestResult(); // This returns HTTP 404
-                return Ok(await agent.GetVariable(id));
+                return Ok(agent.GetVariableWithUnits(id));
             }
             catch (Exception ex)
             {
                 return await HandleExceptionAsync(ex);
             }
         }
-        
+
+        [HttpPost(Name = "Add Variable")]
+        [Authorize(Policy = Policy.AdminOnly)]
+        [APIDescription(type = DescriptionType.e_link, Description = "/Docs/Variables/Add.md")]
+        public async Task<IActionResult> Post([FromBody] VariableWithUnit entity)
+        {
+            try
+            {
+                if (!isValid(entity)) return new BadRequestResult(); // This returns HTTP 404
+
+                Variable tempV = new Variable
+                {
+                    UnitTypeID = entity.UnitTypeID,
+                    Comments = "Default unit"
+                };
+
+                VariableType tempVT = new VariableType
+                {
+                    Name = entity.Name,
+                    Code = entity.Code,
+                    Description = entity.Description
+                };
+
+                var temp = await shared.Add(tempVT);
+                tempV.VariableTypeID = temp.ID;
+                await agent.Add(tempV);
+
+                return Ok(temp);
```

Comment:
We should include the unit type id in the post and put returns since we're sending it back in the get (and to show it went through successfully).

Reply:
Post and Put now return the unit type id.